src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerBase.java

```diff
@@ -773,6 +808,21 @@ protected void _replaceProperty(BeanPropertyMap props, SettableBeanProperty[] cr
                     contextual = contextual.withIgnoreAllUnknown(true);
                 }
             }
+            JsonIncludeProperties.Value inclusions = intr.findPropertyInclusions(accessor);
+            if (inclusions != null) {
+                Set<String> included = inclusions.getIncluded();
+                Set<String> prev = contextual._includableProps;
+                if (prev != null && included != null) {
+                    Set<String> newIncluded = new HashSet<>();
+                    // Make the intersection with the previously included properties.
```

Comment:
Not 100% sure if this is the right default: seems like there would be 3 possibilities:

1. Intersection (must be in class defaults/annotations AND property)
2. Union (add anything property defines)
3. Overwrite/replace (use whatever property defines)

Now, adding "mode" or something in `JsonIncludeProperties` (and `.Value`) would allow specifying these modes easily if we want to. But even without adding that, I wonder what the most likely expected use case would be?
My guess would have been either (3) or (2).